contracts/VaultV1.sol

```diff
@@ -12,51 +12,36 @@ contract VaultV1 is ERC4626, Ownable {
     using SafeTransferLib for ERC20;
 
     //the underlying token the vault accepts
-    ERC20 public immutable UNDERLYING;
+    ERC20 public immutable underlying;
     //the strategy address
-    IStrategy public STRATEGY;
-    //the buffer amount (indicates to bot the minimun amount of tokens that can be stored in the vault)
-    uint256 public BUFFER;
+    IStrategy public strategy;
 
     // Withdraw locking params
     mapping(address => uint256) public requestedWithdraws;
     mapping(address => uint256) public unlockTime;
     uint256 public totalRequestedWithdraws;
     uint256 public withdrawWindow = 24 hours;
 
-    constructor(ERC20 underlying, address strategy) ERC4626(underlying, "TracerVault", "TVLT") {
-        UNDERLYING = ERC20(underlying);
-        STRATEGY = IStrategy(strategy);
+    constructor(ERC20 _underlying, address _strategy) ERC4626(_underlying, "TracerVault", "TVLT") {
+        underlying = ERC20(_underlying);
+        strategy = IStrategy(_strategy);
     }
 
     function totalAssets() public view override returns (uint256) {
         // account for balances outstanding in bot/strategy, check balances
-        return UNDERLYING.balanceOf(address(this)) + STRATEGY.value();
+        return underlying.balanceOf(address(this)) + strategy.value();
     }
 
     //sets the strategy address to send funds
     //Funds get sent to strategy address(controlled by bot)
-    function setStrategy(address strategy) public onlyOwner returns (bool) {
-        STRATEGY = IStrategy(strategy);
-        return true;
-    }
-
-    //sets the buffer amount to leave in vault
-    //Bot queries vault Balance and compares to buffer amount
-    function setBuffer(uint256 buffer) public onlyOwner returns (bool) {
-        BUFFER = buffer;
-        return true;
-    }
-
-    //returns current balance of underlying in vault (represents buffer amount)
-    function bufBalance() public view returns (uint256) {
-        return UNDERLYING.balanceOf(address(this));
+    function setStrategy(address _strategy) public onlyOwner {
+        require(strategy.withdrawable() == 0 && strategy.value() == 0, "strategy still active");
```

Comment:
Will there ever be a case where we need to set new strategy but the old one still contains dust?

Reply:
Potentially. The one case I was trying to avoid here is since `withdraw` is only callable by the vault, if you remove the strategy from the vault then funds will be locked.

Reply:
Perhaps we include an onlyOwner withdrawAll function or similar in the strategy ?